README.md

```diff
@@ -125,6 +127,10 @@ When a client establishes a connection to emmy server and starts communicating w
 
 You can stop emmy server by hitting `Ctrl+C` in the same terminal window.
 
+#### Registration keys
+
+Emmy server verifies registration keys provided by clients when initiating the nym generation procedure. A separate server is expected to provide registration keys to clients via another channel (e.g. QR codes on physical person identification) and save the generated keys to a registration database, read by the Emmy server.
+
```

Comment:
We are using registration key for two purposes now:
- by sending registration key user proves that he is entitled to start nym generation protocol (that means registering with an organization) - this is for example when user is physically present somewhere where he is checked whether entitled to be registered (and in turn he gets registration key using QR code verification)
- as a session key - when protocol is comprised of multiple interactive proofs, the user needs to somehow maintain the session

We will always need such keys actually - to maintain the session. By always I mean in all schemes, but right now we only have one scheme - pseudonym system (all other are primitives). So whatever other scheme we will implement, such verification of key will be needed.

Maybe we should rename registration key into session key and use some other name for the key that is returned by TransferCredential (something like external session key).

Reply:
OK, so I guess the explanation is pretty much correct.

The keys returned by TransferCredential are currently not used anywhere - they are randomly generated and discarded. I think this should be described when we implement something that uses those keys.

I think that the current naming quite accurately describes the purpose of the keys - registration keys for nym generation and session keys for maintaining the session within an Emmy-protocol or for establishing an external session. The two types of session keys should (when they are implemented) be named differently to avoid confusion, but currently we don't have that separation at all.

Reply:
I think we should rename it now. Note that we already have that separation - nym generation is to be called with "registration key" obtained by registration server in one scenario and it will be called with "session key" obtained by TransferCredential in another scenario.

Reply:
OK, so we have 3 cases where the keys are used:

1) Issued at physical person verification (QR code) -> used in nym generation
2) Issued in transfer credentials -> used in nym generation
3) Issued in transfer credentials -> forwared to external application to establish a session there

We only have the first case implemented now, and we call those "registration keys".
For the 2nd and 3rd case, we only have the generation of keys (they get discarded at that moment and aren't verified anywhere at this point).
In the 3rd case, we currently call the keys "session keys", because they are used for establishing a session in the external application.
The 2nd case are the keys used for maintaining the session of an Emmy-protocol. Therefore, I would agree to name these keys "session keys" and use "external session keys" for the 3rd case.

But it doensn't make sense to me to rename the 1st type keys to "session keys".

I can't agree that we already have the separation you mentioned. Because currently nym generation verifies keys in the redis database - where only the registration keys are (those generated by the registration server and corresponding to a QR code). There's no support yet for the other case - calling nym generation with keys, issued by transfer credential.

Reply:
But why wouldn't we use Redis for the second case? At least that was my assumption. In this case all we need to do is store a key in Redis at the end of TransferCredential.

Reply:
Yes, we could do that, we just haven't discussed it yet. I believe case 2 and 3 from my previous comment are out of scope for this PR.

What do you think about the naming?